### PR TITLE
Adding a TF2 Hvd example and test

### PR DESCRIPTION
**Pull request by vandanavk**
#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by anirudhacharya** [comment]
You had asked me to add resent multi-gpu training with horovod into the example folder here. You still want me to do it?

**Comment by vandanavk** [comment]
> You had asked me to add resent multi-gpu training with horovod into the example folder here. You still want me to do it?

this should be enough for now. I took your mnist example

File: tests/zero_code_change/horovod_tests/tensorflow2/test_keras_fit.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    launch_horovod_job(
+        script_file_path=HOROVOD_TF2_TEST_MNIST_SCRIPT,
+        script_args=mode_args,
+        num_workers=num_workers,
+        config_file_path=path,
+        mode=mode,
+    )
+    tr = create_trial(out_dir)
+    assert len(tr.workers()) == num_workers
+    assert len(tr.tensor_names()) == 14
```
Comment (vandanavk) [inline tests/zero_code_change/horovod_tests/tensorflow2/test_keras_fit.py:62]: TF 2.2 has 13 tensors. Need to add a check here

File: examples/tensorflow2/scripts/tf_keras_fit_hvd.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+def train(model, dataset, epoch):
+    # Horovod: Specify `experimental_run_tf_function=False` to ensure TensorFlow
+    # uses hvd.DistributedOptimizer() to compute gradients.
+    model.compile(
+        loss=tf.losses.SparseCategoricalCrossentropy(),
+        optimizer=opt,
+        metrics=["accuracy"],
+        experimental_run_tf_function=False,
+    )
+
```
Comment (NihalHarish) [inline examples/tensorflow2/scripts/tf_keras_fit_hvd.py:73]: Can we test the behavior by parameterizing the `run_eagerly` and `eager` flags?

Comment (vandanavk) [reply examples/tensorflow2/scripts/tf_keras_fit_hvd.py:73]: due to this issue with TF 2.2 [Reference_1](https://stackoverflow.com/questions/58565913/typeerror-an-op-outside-of-the-function-building-code-is-being-passed-a-graph-t), [Reference_2](https://github.com/tensorflow/probability/issues/620), I can't add non-eager testing at this point.

This issue doesn't occur with TF 2.3.0rc0. I'll create a follow up item to get back to this once TF 2.3 is released.

Comment (vandanavk) [reply examples/tensorflow2/scripts/tf_keras_fit_hvd.py:73]: https://github.com/awslabs/sagemaker-debugger/issues/286

**Pull request merged**